plugins/zoom/icon_zoom/util/api.py

```diff
@@ -1,27 +1,103 @@
+from requests.auth import AuthBase
+from requests import Response
+
 from insightconnect_plugin_runtime.exceptions import PluginException
-import json
-from icon_zoom.util.util import generate_jwt_token
+
+
+class BearerAuth(AuthBase):
+    """
+    Authentication class for Bearer auth
+    """
+
+    def __init__(self, access_token: str):
+        self.access_token = access_token
+
+    def __call__(self, r):
+        r.headers["Authorization"] = f"Bearer {self.access_token}"
+        return r
 
 
 class ZoomAPI:
-    def __init__(self, api_key, secret, logger):
-        self.url = "https://api.zoom.us/v2"
-        self.api_key = api_key
-        self.secret = secret
+    def __init__(self, account_id: str, client_id: str, client_secret: str, logger: Logger):
+        self.api_url = "https://api.zoom.us/v2"
+        self.oauth_url = "https://zoom.us/oauth/token"
+        self.account_id = account_id
+        self.client_id = client_id
+        self.client_secret = client_secret
         self.logger = logger
 
-    def get_user(self, user_id):
-        return self._call_api("GET", f"{self.url}/users/{user_id}")
+        self.oauth_token: Optional[str] = None
+        self.oauth_last_refresh_timestamp: Optional[float] = None
+
+    def refresh_oauth_token_if_needed(self) -> None:
+        """
+        Refreshes OAuth token if needed.
+        :return: None
+        """
+        if self.oauth_last_refresh_timestamp and not self._should_refresh_oauth_token(
+            last_refresh_timestamp=self.oauth_last_refresh_timestamp
+        ):
+            return
+
+        # Refresh should happen, so do the refresh.
+        try:
+            self.logger.info("Making call to get new OAuth token")
+            oauth_token = self.get_oauth_token()
+            now_timestamp = self._get_current_time_epoch()
+        except Exception as error:
+            raise PluginException(
+                cause=f"Unable to refresh OAuth token: {error}.",
+                assistance="Ensure connection credentials are correct.",
+                data=error,
+            )
+
+        self.oauth_token = oauth_token
+        self.oauth_last_refresh_timestamp = now_timestamp
+
+    def get_oauth_token(self) -> str:
+        """
+        Retrieves a server-to-server OAuth token from Zoom
+        :return: OAuth token
+        """
+
+        params = {"grant_type": "account_credentials", "account_id": self.account_id}
+
+        auth = HTTPBasicAuth(self.client_id, self.client_secret)
+
+        self.logger.info("Requesting new OAuth token from Zoom...")
+        response = self._call_api(method="POST", url=self.oauth_url, params=params, auth=auth)
+        try:
+            access_token = response["access_token"]
+        except KeyError:
+            raise PluginException(
+                cause=f"Unable to get access token: {response.get('reason', '')}.",
+                assistance="Ensure your connection configuration is using correct credentials.",
+            )
```

Comment:
Looks good. In the latest version of the plugin SDK there is an `OAuth` client that can be used. It's workaround is to pull and refresh the tokens. After that instead of using the `requests.request` method you can use `oauth.request` which contains `Bearer token` in it's headers 😄.